chart/templates/dag-processor/dag-processor-deployment.yaml

```diff
@@ -210,13 +210,16 @@ spec:
           {{- if .Values.dagProcessor.logGroomerSidecar.args }}
           args: {{- tpl (toYaml .Values.dagProcessor.logGroomerSidecar.args) . | nindent 12 }}
           {{- end }}
-          {{- if .Values.dagProcessor.logGroomerSidecar.retentionDays }}
           env:
+          {{- if .Values.dagProcessor.logGroomerSidecar.retentionDays }}
             - name: AIRFLOW__LOG_RETENTION_DAYS
               value: "{{ .Values.dagProcessor.logGroomerSidecar.retentionDays }}"
             - name: AIRFLOW_HOME
```

Comment:
Like discussed [here](https://github.com/astronomer/airflow/pull/1517#discussion_r1927690672), we should set AIRFLOW_HOME regardless.